skipper/userinfo/userinfo.go

```diff
@@ -0,0 +1,81 @@
+package userinfo
+
+import (
+	"fmt"
+
+	authenticationv1 "k8s.io/api/authentication/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	labels "k8s.io/apimachinery/pkg/labels"
+	rbacv1listers "k8s.io/client-go/listers/rbac/v1"
+)
+
+const (
+	// saPrefix represents the service account prefix in admission requests
+	saPrefix = "system:serviceaccount:"
+
+	clusterRoleKind = "ClusterRole"
+	roleKind        = "Role"
+)
+
+// RoleRefs gets the list of roles and cluster roles for the incoming api-request
+func RoleRefs(rbLister rbacv1listers.RoleBindingLister, crbLister rbacv1listers.ClusterRoleBindingLister, user authenticationv1.UserInfo) (roles []string, clusterroles []string, err error) {
+	roleBindings, err := rbLister.List(labels.Everything())
```

Comment:
I might be wrong, but you're looking at all rolebindings in all namespaces right? 

So let's say we configure the role `admin` as a privileged role.  Let user `bob` have the role `admin` in namespace `foo`. If `bob` makes an action in namespace `bar` the validation will be skipped as `bob` has the role `admin`, just not in namespace `bar`.

That doesn't sound right. I think you should only list rolebindings in the affected namespace

Reply:
Roles are always "fully qualified". This function returns a "foo:admin" and a "bar:admin".

See https://github.com/appuio/appuio-cloud-agent/blob/9262a9671e0c4c0c34c72084b4982b4ae6c930a2/skipper/userinfo/userinfo_test.go#L109

Reply:
Ok I missed that. But isn't there still an issue?

If there is a rolebinding giving `bob` the clusterrole `admin` in namespace `foo`, wouldn't this skip the validation in namespace `bar` too? Is that intended?

Reply:
Expected as in same as Kyverno 😅 I that is the best idea idk 🤷‍♀️

Reply:
Well IMO that just wrong and potentially allows privilege escalation...
We shouldn't port bugs from kyverno policies :see_no_evil:

Reply:
As discussed this introduces another potential security issue. I think we should just completely remove role binding stuff. The "features" is IMO questionable anyway